tools/keepalived/keepalived/check/ipwrapper.c

```diff
@@ -197,16 +197,20 @@ init_service_vs(virtual_server_t * vs)
 			SET_ALIVE(vs);
 	}
 
-	/*Set local ip address in "FNAT" mode of IPVS */
-	if ((vs->loadbalancing_kind == IP_VS_CONN_F_FULLNAT) && vs->local_addr_gname) { 
+	/* Set local ip address in "FNAT" mode of IPVS */
+	if (vs->local_addr_gname &&
+        (vs->loadbalancing_kind == IP_VS_CONN_F_SNAT ||
+         vs->loadbalancing_kind == IP_VS_CONN_F_FULLNAT)) {
```

Comment:
place `IP_VS_CONN_F_FULLNAT` before `IP_VS_CONN_F_SNAT` , for  the latter are of less probability to use than the former.

Reply:
Okay. And I will send the code review against devel branch.